agent/lib_guzzle6.c

```diff
@@ -65,12 +65,15 @@
  */
 #if ZEND_MODULE_API_NO >= ZEND_5_5_X_API_NO
 
-/* {{{ newrelic\Guzzle6\RequestHandler class definition and methods */
-
 /*
  * True global for the RequestHandler class entry.
  */
 zend_class_entry* nr_guzzle6_requesthandler_ce;
+zval* config;
+zend_class_entry* guzzle_client_ce;
+zval* handler_stack;
+zval* middleware = NULL;
+zval* retval;
```

Comment:
These variables were initially global because nr_guzzle6_client_construct and nr_guzzle7_client_construct both used these variables. However, I have added a helper function called nr_guzzle_client_construct_helper, which reduces code duplication and does not require these variables to be global anymore. Fixed! [6a879ff](https://github.com/newrelic/newrelic-php-agent/pull/498/commits/6a879ff10ae7093861b2b3f5893d112b00553284)